com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/HDLightUnitSlider.cs

```diff
@@ -0,0 +1,289 @@
+        {
+            // Convert the internal lumens into the actual light unit value
+            value = LumenToUnit(value);
+            unit = k_UnitNames[(int)m_Unit];
+
+            return base.GetLightUnitTooltip(baseTooltip, value, unit);
+        }
+
+        float UnitToLumen(float value)
+        {
+            if (m_Unit == LightUnit.Lumen)
+                return value;
+
+            // Punctual slider currently does not have any regard for spot shape/reflector.
+            // Conversions need to happen as if light is a point, and this is the only setting that influences that.
+            m_Light.enableSpotReflector.boolValue = false;
+
+            return HDLightUI.ConvertLightIntensity(m_Unit, LightUnit.Lumen, m_Light, m_Editor, value);
+        }
+
+        float LumenToUnit(float value)
+        {
+            if (m_Unit == LightUnit.Lumen)
+                return value;
+
+            // Once again temporarily disable reflector in case we called this for tooltip or context menu preset.
+            m_Light.enableSpotReflector.boolValue = false;
+
+            value = HDLightUI.ConvertLightIntensity(LightUnit.Lumen, m_Unit, m_Light, m_Editor, value);
+
+            // Restore the state of spot reflector on the light.
+            m_Light.enableSpotReflector.boolValue = m_SpotReflectorEnabled;
+
+            return value;
+        }
+
+        protected override void SetValueToPreset(SerializedProperty value, LightUnitSliderUIRange preset)
+        {
+            m_Light?.Update();
+
+            // Convert to the actual unit value.
+            value.floatValue = LumenToUnit(preset.presetValue);
+
+            m_Light?.Apply();
+        }
+    }
+
+    internal class HDLightUnitSliderUIDrawer
+    {
+        static HDPiecewiseLightUnitSlider k_DirectionalLightUnitSlider;
+        static HDPunctualLightUnitSlider  k_PunctualLightUnitSlider;
+        static HDPiecewiseLightUnitSlider k_ExposureSlider;
+
+        static HDLightUnitSliderUIDrawer()
+        {
+            // Maintain a unique slider for directional/lux.
+            k_DirectionalLightUnitSlider = new HDPiecewiseLightUnitSlider(LightUnitSliderDescriptors.LuxDescriptor);
+
+            // Internally, slider is always in terms of lumens, so that the slider is uniform for all light units.
+            k_PunctualLightUnitSlider = new HDPunctualLightUnitSlider(LightUnitSliderDescriptors.LumenDescriptor);
+
+            // Exposure is in EV100, but we load a separate due to the different icon set.
+            k_ExposureSlider = new HDPiecewiseLightUnitSlider(LightUnitSliderDescriptors.ExposureDescriptor);
+        }
+
+        // Need to cache the serialized object on the slider, to add support for the preset selection context menu (need to apply changes to serialized)
+        // TODO: This slider drawer is getting kind of bloated. Break up the implementation into where it is actually used?
```

Comment:
is this TODO tracked as a JIRA somewhere?

Reply:
not that I know